smartsim/_core/control/controller.py

```diff
@@ -296,21 +364,38 @@ def _launch(self, manifest: Manifest) -> None:
                 msg = "Attempted to launch a second Orchestrator instance. "
                 msg += "Only 1 Orchestrator can be active at a time"
                 raise SmartSimError(msg)
-            self._launch_orchestrator(orchestrator)
+            self._launch_orchestrator(orchestrator, launched_manifest)
```

Comment:
What do you think of having `._launch_xxx` methods return something like a ``LaunchManifest`` (basically, a little event type that contains the details) instead of modifying the launched_manifest in-place? Maybe that gives us an easier way to test that the right things are changed and minimizes the ability to change the launch manifest in undesired ways?

Reply:
Interesting! I tried decoupling the construction of the `LaunchedManifest` by moving it into a `LuanchedManifestBuilder` class which at least makes the intention of the code clear, but I'm still not happy with the out parameter.

I might need to pick you brain on this as I would much rather keep the use of out parameters to minimum and constructing piece wise via "actions" seems like it could be a better solution!!